### PR TITLE
add MultiStepLR; correct CosineDecay according to the original paper and pytorch

### PR DESCRIPTION
**Pull request by Edenzzzz**
This is for better academic reproducibility and alignment with Pytorch to make it more user-friendly. 
1. Add MultiStepLR which decays lr by milestones, as in Pytorch.
2. Correct the implementation of CosineDecay. In the original paper(https://arxiv.org/abs/1608.03983), if you check the formulation it is clear that the decay is epoch-wise, as in Pytorch. Making it batch-wise as now can lead to non-reproducible results for developers who switch between Pytorch and TF As most research papers now are in Pytorch, this bug will deter researchers from implementing/reimplementing methods in Tensorflow.
However for compatibility with old code, this argument is optional, so if you don't change anything it will be the old batch-wise decay.

### CODE REVIEW

**Comment by qlzh727** [comment]
Please remove all the unrelated files from this PR.

**Comment by Edenzzzz** [comment]
> Please remove all the unrelated files from this PR.

Could you please identify those? I didn't add files.

**Comment by qlzh727** [comment]
Please note that there are a bunch of files in this PR are just format change, which is not related to this PR.

Please remove them.

File: .vscode/settings.json
Comment (rchao) [inline .vscode/settings.json:1]: Thanks for the PR! Is the deletion of this file intended?

Comment (Edenzzzz) [reply .vscode/settings.json:1]: Yes! My VScode automatically added this file..

Comment (rchao) [reply .vscode/settings.json:1]: It seems this file is intended to be here [history](https://github.com/keras-team/keras/commits/master/.vscode/settings.json). Can you remove this deletion from your PR?

Comment (Edenzzzz) [reply .vscode/settings.json:1]: Done. My mistake.

Comment (rchao) [reply .vscode/settings.json:1]: There appears to be still some changes done on this file. Can you revert them?

Comment (Edenzzzz) [reply .vscode/settings.json:1]: Done. Seems that copying it back doesn't do the job; had to git update-index --assume-unchanged

**Comment by Edenzzzz** [comment]
Anything I need to do at this point?

**Comment by rchao** [comment]
We're waiting for the internal tests to finish running. (please change the tag to finish running once done, thanks!)

**Comment by sachinprasadhs** [comment]
Hi, 

Tests are failing with below error message   ```
File "/build/work/98b77b6d908e91741657894b0cf073ac979f/google3/runfiles/google3/third_party/tensorflow/python/util/module_wrapper.py", line 248, in _getattr
    attr = getattr(self._tfmw_wrapped_module, name)
AttributeError: module 'tensorflow.compat.v2' has no attribute 'keras'
```

**Comment by Edenzzzz** [comment]
> Hi,
> 
> Tests are failing with below error message
> 
> ```
> File "/build/work/98b77b6d908e91741657894b0cf073ac979f/google3/runfiles/google3/third_party/tensorflow/python/util/module_wrapper.py", line 248, in _getattr
>     attr = getattr(self._tfmw_wrapped_module, name)
> AttributeError: module 'tensorflow.compat.v2' has no attribute 'keras'
> ```

I haven't modified any part of this folder, so the problem should lie in your code?

**Comment by Edenzzzz** [comment]
It's hard to infer where this error occured as this is only a wrapper that doesn't exist in my directory. Could you please provide more information?

**Comment by Edenzzzz** [comment]
@rchao Could you please provide more context? It appears to be a package versioning issue with the test environment.

**Comment by Edenzzzz** [comment]
@sachinprasadhs could you provide some further comments?

**Comment by gbaned** [comment]
Hi @rchao Any update on this PR? Please. Thank you!

**Comment by haifeng-jin** [comment]
Rick may not be able to further assist with this PR. Let's assign it again in our next triage.

**Comment by sachinprasadhs** [comment]
Hello, Thank you for submitting a pull request.

We're currently in the process of migrating the new `Keras 3 code base` from `keras-team/keras-core` to `keras-team/keras`. 
Consequently, merging this PR is not possible at the moment. After the migration is successfully completed, feel free to reopen this PR at `keras-team/keras` if you believe it remains relevant to the `Keras 3 code base`. If instead this PR fixes a bug or security issue in legacy `tf.keras`, you can instead reopen the PR at `keras-team/tf-keras`, which hosts the `TensorFlow-only`, legacy version of `Keras`.

**Pull request closed without merging**